UM/Qt/qml/UM/CheckBox.qml

```diff
@@ -45,21 +50,39 @@ CheckBox
             states: [
                 State {
                     name: "unchecked"
-                    when: control.enabled && control.checkState == Qt.Unchecked
-                    PropertyChanges { target: indicator_item; width: Math.round(parent.width / 1.5); height: width }
-                    PropertyChanges { target: indicator_item; color: UM.Theme.getColor("checkbox_mark"); source: UM.Theme.getIcon("Check", "low"); opacity: false; }
+                    when: control.checkState == Qt.Unchecked
+                    PropertyChanges
+                    {
+                        target: indicator_item
+                        width: Math.round(parent.width / 1.5)
+                        height: width
+                        visible: false
+                        source: ""
+                    }
                 },
                 State {
                     name: "partially_checked"
-                    when: control.enabled && control.checkState == Qt.PartiallyChecked
-                    PropertyChanges { target: indicator_item; width: Math.round(parent.width / 1.8); height: width  }
-                    PropertyChanges { target: indicator_item; color: UM.Theme.getColor("checkbox_border"); source: UM.Theme.getIcon("Solid"); opacity: true; }
+                    when: control.checkState == Qt.PartiallyChecked
+                    PropertyChanges
+                    {
+                        target: indicator_item
+                        width: Math.round(parent.width / 1.8)
+                        height: width
```

Comment:
I'm fairly sure that you don't need to set it again. When it's initially set, it creates a binding.